pocs/filterwheel/zwo.py

```diff
@@ -0,0 +1,173 @@
+                         timeout=timeout,
+                         serial_number=serial_number,
+                         *args, **kwargs)
+
+        if FilterWheel._driver is None:
+            # Initialise the driver if it hasn't already been done
+            FilterWheel._driver = EFWDriver(library_path=library_path)
+
+        self._device_name = device_name
+        self.connect()
+        self.is_unidirectional = unidirectional
+        if initial_filter:
+            self.move_to(initial_filter, blocking=True)
+
+    def __del__(self):
+        """Attempt some clean up."""
+        with suppress(AttributeError):
+            device_name = self._device_name
+            FilterWheel._assigned_filterwheels.discard(device_name)
+            self.logger.debug('Removed {} from assigned filterwheels list'.format(device_name))
+
+##################################################################################################
+# Properties
+##################################################################################################
+
+    @AbstractFilterWheel.position.getter
+    def position(self):
+        """ Current integer position of the filter wheel """
+        return self._driver.get_position(self._handle) + 1  # 1-based numbering
+
+    @property
+    def is_moving(self):
+        """ Is the filterwheel currently moving """
+        return not self._move_event.is_set()
+
+    @property
+    def is_unidirectional(self):
+        return self._driver.get_direction(self._handle)
+
+    # ZWO filterwheels can be set to be unidirectional or bidirectional.
+    @is_unidirectional.setter
+    def is_unidirectional(self, unidirectional):
+        self._driver.set_direction(self._handle, bool(unidirectional))
+
+##################################################################################################
+# Methods
+##################################################################################################
+
+    def connect(self):
+        """Connect to filter wheel."""
+        # Scan for connected filterwheels. This needs to be done at least once before
+        # opening a connectio.
+        FilterWheel._filterwheels = self._driver.get_devices()
+
+        if len(FilterWheel._filterwheels) == 0:
+            msg = "No ZWO filterwheels found."
+            self.logger.error(msg)
+            raise error.NotFound(msg)
+
+        if self._device_name:
+            self.logger.debug(f"Looking for filterwheel with device name '{self._device_name}'.")
+            if self._device_name not in FilterWheel._filterwheels:
+                msg = f"No filterwheel with device name '{self._device_name}'."
+                self.logger.error(msg)
+                raise error.NotFound(msg)
+            if self._device_name in FilterWheel._assigned_filterwheels:
+                msg = f"Filterwheel '{self._device_name}' already in use."
+                self.logger.error(msg)
+                raise error.PanError(msg)
```

Comment:
Should this (and below) really raise an error? Would be nice if `connect` where idempotent.

Reply:
I think so. This behaviour is the same as most of the other hardware classes (SBIG, FLI & ZWO cameras, & Birger focusers).  The intent is to prevent a situation where the wrong device is being controlled, and I think its reasonable to raise an error if seems like that's about to occur.

One thing I have done here is add at check at the start of `connect()` to log a warning and skip all this if the filterwheel is already 'connected', as an extra attempt to connect shouldn't really result in an error.